rainier-core/src/main/scala/com/stripe/rainier/core/Multivariate.scala

```diff
@@ -0,0 +1,50 @@
+package com.stripe.rainier.core
+
+import com.stripe.rainier.compute._
+
+case class MVNormal(k: Int, cov: Covariance) extends Distribution[Seq[Double]] {
+ def translate(location: Vec[Real]): Distribution[Seq[Double]] = ???
+
+ def logDensity(seq: Seq[Seq[Double]]) =
+    Vec.from(seq).map(logDensity).columnize
+
+  def logDensity(x: Vec[Real]) = 
+    ((Real.Pi * 2).log + 
+    cov.logDeterminant + 
+    x.dot(cov.inverseMultiply(x))) / -2
+
+  def generator = {
+    val iidNormals = Normal.standard.generator.repeat(k).map(_.toArray)
+    cov.choleskyGenerator.zip(iidNormals).map{
+     case (a, z) => Cholesky.lowerTriangularMultiply(a, z)
+    }
+  }    
+}
+
+object MVNormal {
+    def eta(k: Int, eta: Real): MVNormal = ???
+    def eta(k: Int, eta: Real, sigma: Continuous): MVNormal = ???
+    def rho(k: Int, rho: Map[(Int,Int),Real]): MVNormal = ???
+    def rho(k: Int, rho: Map[(Int,Int),Real], sigma: Continuous): MVNormal = ???
+}
+
+trait Covariance {
```

Comment:
should this have a size? is that not ever needed?

Reply:
It should, yeah, so that we can `require` that it be the same as the `MVNormal`'s size.